src/transformStateWithClones.js

```diff
@@ -65,7 +65,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, transforms) {
-  // write code here
+  const trueCopy = { ...state };
+  const result = [];
+
+  for (const transform of transforms) {
+    if (transform.operation === 'addProperties') {
+      Object.assign(trueCopy, transform.properties);
+      result.push({ ...trueCopy });
+    }
+
+    if (transform.operation === 'removeProperties') {
+      for (const prop of transform.properties) {
+        if (trueCopy.hasOwnProperty(prop)) {
+          delete trueCopy[prop];
+        }
+      }
+      result.push({ ...trueCopy });
+    }
+
+    if (transform.operation === 'clear') {
+      for (const key in trueCopy) {
+        delete trueCopy[key];
+      }
+      result.push({ ...trueCopy });
```

Comment:
you wrote result.push 3 times, but you can push only once at the end of each iteration